clang/include/clang/AST/DeclTemplate.h

```diff
@@ -1008,6 +1008,15 @@ class FunctionTemplateDecl : public RedeclarableTemplateDecl {
     return getTemplatedDecl()->isThisDeclarationADefinition();
   }
 
+  bool isCompatibleWithDefinition() const {
+    return getTemplatedDecl()->isInstantiatedFromMemberTemplate() ||
+           isThisDeclarationADefinition();
+  }
+  void setInstantiatedFromMemberTemplate(FunctionTemplateDecl *D) {
+    getTemplatedDecl()->setInstantiatedFromMemberTemplate();
+    RedeclarableTemplateDecl::setInstantiatedFromMemberTemplate(D);
+  }
```

Comment:
So the flag on FunctionDecl now appears to be just a shortcut of getDescribedTemplate()->getInstantiatedFromMemberTemplate(), or am I missing something else?

Reply:
getInstantiatedFromMemberTemplate() uses the common area, which means whenever you set it for one declaration, it's set the same for all its redeclarations.
But the flag does not use the common area, it's a separate flag per redeclaration.

Reply:
Get it, thanks!